app/analyzers/sudden_appearance.py

```diff
@@ -0,0 +1,144 @@
+
+    def __init__(self, model_name, config_section):
+        super(SuddenAppearanceAnalyzer, self).__init__("sudden_appearance", model_name, config_section)
+
+    def _extract_additional_model_settings(self):
+        """
+        Override method from Analyzer
+        """
+        self.model_settings["timestamp_field"] = settings.config.get("general",
+                                                                     "timestamp_field",
+                                                                     fallback="@timestamp")
+
+        self.model_settings["target"] = self.config_section["target"].replace(' ', '').split(",")
+        self.model_settings["aggregator"] = self.config_section["aggregator"].replace(' ', '').split(",")
+
+        self.model_settings["slide_window_days"] = self.config_section.getint("slide_window_days")
+        self.model_settings["slide_window_hours"] = self.config_section.getint("slide_window_hours")
+        self.model_settings["slide_window_mins"] = self.config_section.getint("slide_window_mins", fallback=0)
+        self.model_settings["slide_jump_days"] = self.config_section.getint("slide_jump_days")
+        self.model_settings["slide_jump_hours"] = self.config_section.getint("slide_jump_hours")
+        self.model_settings["slide_jump_mins"] = self.config_section.getint("slide_jump_mins", fallback=0)
+
+        self.model_settings["trigger_slide_window_proportion"] = self.config_section.getfloat(
+            "trigger_slide_window_proportion")
+
+        self.delta_history_win = dt.timedelta(days=self.model_settings["history_window_days"],
+                                              hours=self.model_settings["history_window_hours"])
+
+        self.delta_slide_win = dt.timedelta(days=self.model_settings["slide_window_days"],
+                                            hours=self.model_settings["slide_window_hours"],
+                                            minutes=self.model_settings["slide_window_mins"])
+
+        self.jump_win = dt.timedelta(days=self.model_settings["slide_jump_days"],
+                                     hours=self.model_settings["slide_jump_hours"],
+                                     minutes=self.model_settings["slide_jump_mins"])
+
+        if self.delta_history_win < self.delta_slide_win:
+            raise ValueError("Slide window of size %s is bigger than history window of size %s" % (
+                self.delta_slide_win, self.delta_history_win))
+
+        if self.delta_slide_win < self.jump_win:
+            raise ValueError("Jump window of size %s is bigger than slide window of size %s" % (
+                self.jump_win, self.delta_slide_win))
+
+        if self.jump_win == dt.timedelta(days=0, hours=0, minutes=0):
+            raise ValueError("Jump window has no size.")
+
+        if self.delta_slide_win == dt.timedelta(days=0, hours=0, minutes=0):
+            raise ValueError("Slide window has no size.")
+
+    def evaluate_model(self):
```

Comment:
If this is only used for unit testing, then it should be removed here and added to the unit test itself. You can instantiate an object and assign a new method to it on the fly if needed.